java/memory/memory-core/src/main/java/org/apache/arrow/memory/BufferAllocator.java

```diff
@@ -126,6 +134,30 @@ BufferAllocator newChildAllocator(
    */
   long getHeadroom();
 
+  /**
+   * Forcibly allocate bytes. Returns whether the allocation fit within limits.
+   *
+   * @param size to increase
+   * @return Whether the allocation fit within limits.
+   */
+  boolean forceAllocate(long size);
```

Comment:
I didn't see where this or releaseBytes are used in code. What is the purpose of releaseBytes and of makign forceAllocate public

Reply:
Hi @rymurr,

E.g. In AllocationMangager code, we were calling BaseAllocator#releaseBytes:

https://github.com/apache/arrow/blob/d9525f1b19c00e440b3af83a68d1f2bba89a0a24/java/memory/memory-core/src/main/java/org/apache/arrow/memory/AllocationManager.java#L155

After changes in this PR, we are now trying to call BufferAllocator#releaseBytes:

https://github.com/apache/arrow/pull/7768/files#diff-68d17bac8aafb7004e2fb310f131df5fR160

That's why the method is added to BufferAllocator.

Reply:
:+1:  sorry, missed that